pkg/frontend/transport/handler.go

```diff
@@ -122,6 +124,16 @@ func NewHandler(cfg HandlerConfig, roundTripper http.RoundTripper, log log.Logge
 			Help: "Number of samples fetched to execute a query.",
 		}, []string{"user"})
 
+		h.queryTotalQueryableSamples = promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "cortex_query_total_queryable_samples_total",
+			Help: "Number of total queryable samples to execute a query.",
+		}, []string{"user"})
+
+		h.queryPeakSamples = promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "cortex_query_peak_samples_total",
+			Help: "Highest count of samples considered to execute a query.",
```

Comment:
Can you please explain the usecase of tracking peak samples using a counter metric?

Reply:
I think like promQL: `sum by (user) (irate(cortex_query_peak_samples_total[5m]))`, the cortex user can track the rate of how many samples are loaded in memory for each tenant's overall queries.

One usecase is: 
- The cortex user can track query memory usage (used by loaded samples) for each tenant and turn promQL to reduce query memory usage.

Reply:
OK. For a counter metric I feel it is more useful to just track total samples, not peak samples.
Would it be more useful to use native histogram instead?

Reply:
Yes, I think histogram is the best.